### PR TITLE
search pagination

### PR DESCRIPTION
**Pull request by alxnddr**
NOTE: Front-end code is ready for review, waiting for an adjustment on the back end. Current total is limited to 50 so you won't be able to use the pagination unless changing `PAGE_SIZE` in `frontend/src/metabase/home/containers/SearchApp.jsx` for testing purposes.

#### Description

Add pagination to the global search (enter query and press enter).

Screen Shot 2021-05-05 at 23 11 45

### CODE REVIEW

**Comment by alxnddr** [comment]
@kdoh @howonlee 
Previously, on the right side, we showed available filters based on items we received from the server. It is not exactly correct because a shorter query can contain lesser different types of items than a longer one.

### Example
`or` returned 50 search results that contain only 3 types of items:
Screen Shot 2021-05-05 at 02 35 57

But this does not mean that there are no other types of items that match `or` query. As an example, if we narrow the search to `ord` we will receive more types of items, and therefore will show more filters:

Screen Shot 2021-05-05 at 02 31 21

To make it work correctly, we can permanently show all filters and some of them will return 0 results. Or we can return a list of available item types from the back end for a specified query, but it does not sound good to me.

What do you think? Any other ideas?

File: src/metabase/search/config.clj [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 (def ^:const max-filtered-results
   "Number of results to return in an API response"
-  50)
+  125)
```
Comment (alxnddr) [inline src/metabase/search/config.clj:25]: @kdoh @howonlee it looks like due to performance reasons we will have 2.5 pages at a maximum which does not bring much value. Can we go with a higher number here or maybe we should just render all 125 items without pagination?

Comment (howonlee) [reply src/metabase/search/config.clj:25]: I think the pagination was all for UX reasons not perf

I tried 1000 with 1000 search reusults and that was annoyingly slow, so I just pulled 125 out of my butt, it's not like a principled rigorous profile or anything

File: frontend/test/metabase/scenarios/search/search.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+const generateQuestions = count => {
+  _.range(count).map(i =>
+    cy.createQuestion({
+      name: `generated question ${i}`,
+      query: {
+        "source-table": ORDERS_ID,
+        aggregation: [["count"]],
+        breakout: [
+          ["field", ORDERS.CREATED_AT.id, { "temporal-unit": "hour-of-day" }],
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/search/search.cy.spec.js:53]: Wait, wait... what is `ORDERS.CREATED_AT.id`? The `CREATED_AT` part **is** an id (12).

```suggestion
          ["field", ORDERS.CREATED_AT, { "temporal-unit": "hour-of-day" }],
```

File: src/metabase/search/config.clj (hunk 1 above)
Comment (kdoh) [reply src/metabase/search/config.clj:25]: Sorry I may be a bit confused, what is this number doing here? And just for proper context, our primary motivation for adding pagination around the app is definitely for performance reasons.

**Comment by kdoh** [comment]
@alxnddr r.e. your question about the item filters on the side, I think I'd want to try and avoid giving the user options that when they click on them provide no value (e.x. have an item that says dashboards, click on it, and there aren't any). I'm not sure what the technical implications of that are, but that's my UX preference.

File: src/metabase/search/config.clj (hunk 1 above)
Comment (alxnddr) [reply src/metabase/search/config.clj:25]: @kdoh even with pagination, we have to limit the total number of matched items due to performance reasons. No matter if we request only the first page with 50 of 10000 items it is still slow on the back end. Is that correct, @howonlee?

**Comment by alxnddr** [comment]
@kdoh to implement it properly, we need the back end to return available types of items for a specified query. It might hurt the performance. Let's wait for @howonlee on this

File: frontend/test/metabase/scenarios/search/search.cy.spec.js (hunk 2 above)
Comment (alxnddr) [reply frontend/test/metabase/scenarios/search/search.cy.spec.js]: Thanks for a good catch, fixed it!

File: src/metabase/search/config.clj (hunk 1 above)
Comment (kdoh) [reply src/metabase/search/config.clj:25]: Got it. I'd see if we could up this limit a bit then.

File: frontend/src/metabase/components/PaginationControls.jsx [hunk 3]
```diff
@@ -8,25 +8,37 @@ export default function PaginationControls({
...
+
   return (
-    <div className="flex align-center" aria-label="pagination">
-      <span className="text-bold mr1">
+    <div className="flex align-center text-bold" aria-label="pagination">
+      <span className="mr1">
         {page * pageSize + 1} - {page * pageSize + itemsLength}
+        {showTotal && (
+          <React.Fragment>
+            <span className="text-light">&nbsp;of&nbsp;</span>
```
Comment (daltojohnso) [inline frontend/src/metabase/components/PaginationControls.jsx:26]: not the most significant line of text in the app, but I suppose we should localize it

**Review by daltojohnso — CHANGES_REQUESTED** [review]
looking good -- just a missed localization of `of`

File: frontend/src/metabase/components/PaginationControls.jsx (hunk 3 above)
Comment (howonlee) [reply frontend/src/metabase/components/PaginationControls.jsx:26]: prepositions are actually a real pain in the butt for some languages

**Review by daltojohnso — APPROVED** [review]

**Comment by howonlee** [comment]
so part of the half-assedness of my half-assed profiling was that I was testing and not splitting apart network roundtrip from the actual calculations, and it does seem network dominates. (if you think about this this was _extremely_ half-assed profiling, cuz a large part of the point of the pagination is to squeeze down network time, lol) so it piddles away for 1000 results at 500ms on my laptop for the actual query itself, which is very reasonable

**Comment by howonlee** [comment]
memorializing internal convo: yes we can return the types but they have to be separate queries to not take a million years. those new queries are incoming probably in another PR. but we can jack up the 125 figure for cardinality of filtered results to 1000, network hop dominates on that one

**Review by nemanjaglumac — APPROVED** [review]
Ran tests locally. All passing :)

**Comment by alxnddr** [comment]
@howonlee it does seem like CI failures are legit, can you please check?

**Review by kdoh — APPROVED** [review]
Looks good.

**Pull request merged**